### ISSUE TITLE
Azure DevOps Integration Failing Sometimes

### ISSUE THREAD

**Issue by garoyeri** [fixie/fixie#209]
## Steps to Reproduce

1. Setup Fixie on a "classic" build pipeline in Azure DevOps following the instructions (allow access to OAuth token).
2. Run unit tests either automated or manually

## Expected Behavior

Tests pass because my code is awesome and has no bugs.

## Actual Behavior

Sporadically, tests are marked as failed with this exception trace:
```
System.AggregateException
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Fixie.Internal.Listeners.AzureListener.Post(HttpClient client, String uri, String mediaType, String content) in C:\dev\fixie\src\Fixie\Internal\Listeners\AzureListener.cs:line 144
   at Fixie.Internal.Listeners.AzureListener.Post(CaseCompleted message, Action`1 customize) in C:\dev\fixie\src\Fixie\Internal\Listeners\AzureListener.cs:line 124
   at Fixie.Internal.Bus.Publish[TMessage](TMessage message) in C:\dev\fixie\src\Fixie\Internal\Bus.cs:line 22
   at Fixie.Internal.ClassRunner.<>c__DisplayClass6_0.<Run>b__0(Action`1 caseLifecycle) in C:\dev\fixie\src\Fixie\Internal\ClassRunner.cs:line 96
   at Fixie.TestClass.RunCases(Action`1 caseLifecycle) in C:\dev\fixie\src\Fixie\TestClass.cs:line 59
   at SKF.DigitalPlatform.Api.Tests.TestingConvention.Execute(TestClass testClass) in /home/vsts/work/1/s/src/SKF.DigitalPlatform.Api.Tests/TestingConvention.cs:line 32
   at Fixie.Internal.ClassRunner.Run(Type testClass, Boolean isOnlyTestClass) in C:\dev\fixie\src\Fixie\Internal\ClassRunner.cs:line 107

------- Inner Exception: System.Net.Http.HttpRequestException -------
An error occurred while sending the request.
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithNtConnectionAuthAsync(HttpConnection connection, HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithRetryAsync(HttpRequestMessage request, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.FinishSendAsyncBuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)

------- Inner Exception: System.IO.IOException -------
Unable to read data from the transport connection: Connection reset by peer.
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.GetResult(Int16 token)
   at System.Net.Security.SslStream.<FillBufferAsync>g__InternalFillBufferAsync|215_0[TReadAdapter](TReadAdapter adap, ValueTask`1 task, Int32 min, Int32 initial)
   at System.Net.Security.SslStream.ReadAsyncInternal[TReadAdapter](TReadAdapter adapter, Memory`1 buffer)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)

------- Inner Exception: System.Net.Sockets.SocketException -------
Connection reset by peer
```
Rerunning the test succeeds without issue.

**Comment by plioi** [comment]
For context, ADO exposes an API to the process, where we post test results. We do so here: https://github.com/fixie/fixie/blob/develop/src/Fixie/Internal/Listeners/AzureListener.cs#L137

"Connection reset by peer"... Trying to think how much of that is under our control. Does this happen only after a long test run, in which our token might have timed out? Or does it sometimes happen early in the test run? I also wonder if we're flooding with too many requests. Might be possible to batch up results one test class at a time (our xml listener has to do similar).

**Comment by plioi** [comment]
In the meantime, consider removing the token configuration so you can at least get the text-mode results un-obfuscated by this.

**Comment by garoyeri** [comment]
Unfortunately, the last time I ran this, I didn't save the full Azure DevOps logs, only the exception trace, so I don't have a good answer for you on the timing or the token.

I agree that batching might get a better result that might reduce the chance of issues. This issue doesn't happen often at all (maybe once a month if I see it). It could be solved with just a simple retry.

**Comment by plioi** [comment]
I believe we've addressed stability in [v2.2.1](https://github.com/fixie/fixie/releases/tag/v2.2.1). See #211 for a full rundown of what we're doing now. If this surfaces again, let's discuss that in a new Issue since we'll have more interesting evidence in the captured console log.
